javascript/src/js/app.js

```diff
@@ -0,0 +1,412 @@
+/**
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Library General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+angular
+.module('olaApp', ['ngRoute'])
+.factory('$ola', ['$http', function ($http) {
+  var postEncode = function(data) {
```

Comment:
Is there not something built in to do this? It also looks like it will fail if the universe name has an equals or ampersand in it for example (although the current code actually fails with an ampersand and truncates the ampersand and everything to the right of it, so that may be server side).